src/pointer.rs

```diff
@@ -121,10 +125,21 @@ impl Pointer {
     /// Attempts to parse a string into a `Pointer`.
     ///
     /// If successful, this does not allocate.
+    ///
+    /// ## Errors
+    /// Returns a `ParseError` if the string is not a valid JSON Pointer.
     pub fn parse<S: AsRef<str> + ?Sized>(s: &S) -> Result<&Self, ParseError> {
         validate(s.as_ref()).map(Self::new)
     }
-
+    /// Attempts to parse a string into a `Pointer`.
+    ///
+    /// If successful, this does not allocate.
+    ///
+    /// ## Panics
+    /// Panics string is not a valid JSON Pointer.
+    pub fn must_parse<S: AsRef<str> + ?Sized>(s: &S) -> &Self {
```

Comment:
This is very similar to `Pointer::from_static`. I'm not opposed to adding it, but at least in the tests I think you can use that instead.

Reply:
Ya, it is. I wanted it on `PointerBuf` and copied it over to `Pointer` for uniformity. I'm cool with dumping either or both.

Reply:
I think it's ok to keep both, we should just favour `from_static` because that gives a more convenient lifetime.

Reply:
I'm cool with dumping the `must_parse` methods, I don't recall running into too many of them in the wild. Where as `from_static` being `const fn` is awesome and provides a bit more utility than avoiding a call to `unwrap`/`expect`.